### PR TITLE
[jsk_pcl_ros] set tracking model with marker in particle_filter_tracker

### PR DESCRIPTION
**Pull request by aginika**
DONT MERGE YET

### CODE REVIEW

**Comment by k-okada** [comment]
please add example or documents too

--
◉ Kei Okada

**Comment by aginika** [comment]
@k-okada 
Thank you. I added sample and documents.

I also move the target according to the marker.position
and
I added color too.

Now, user can execute demonstration with below code if openni topic and tf is correctly published.
```
roslaunch jsk_pcl_ros tracking.launch _track_with_model_sample:=true
```

**Comment by aginika** [comment]
Please merge if this passed. 
Thanks

File: jsk_pcl_ros/src/particle_filter_tracking_nodelet.cpp [hunk 1]
```diff
@@ -394,6 +398,31 @@ namespace jsk_pcl_ros
...
+    if(marker.type == visualization_msgs::Marker::TRIANGLE_LIST && !marker.points.empty()){
+      ROS_INFO("Reset Tracker Model with renew_model_with_marker_topic_cb");
+      pcl::PointCloud<PointT>::Ptr cloud(new pcl::PointCloud<PointT>);
+      markerMsgToPointCloud(marker,
+                            marker_to_pointcloud_sampling_nums_,
+                            *cloud
+                            );
+
+      Eigen::Affine3d trans_3d; Eigen::Affine3f trans;
+      tf::poseMsgToEigen(marker.pose, trans_3d);
```
Comment (garaemon) [inline jsk_pcl_ros/src/particle_filter_tracking_nodelet.cpp]: if you include pcl_conversion_util.h, you can use tf::poseMsgToEigen with Eigen::Affine3f

File: jsk_pcl_ros/src/particle_filter_tracking_nodelet.cpp [hunk 2]
```diff
@@ -394,6 +398,31 @@ namespace jsk_pcl_ros
...
+      trans = (Eigen::Affine3f) trans_3d;
+      pcl::transformPointCloud(*cloud, *cloud, trans);
+
+      frame_id_ = marker.header.frame_id;
+      reset_tracking_target_model(cloud);
+    }else{
+      ROS_ERROR(" Marker Models type is not TRIANGLE ");
```
Comment (garaemon) [inline jsk_pcl_ros/src/particle_filter_tracking_nodelet.cpp]: Use `JSK_ROS_ERROR`

**Comment by aginika** [comment]
@garaemon Thank you for your advise.
I modified code.

**Pull request merged**